include/public/osqp_api_functions.h

```diff
@@ -259,15 +259,20 @@ OSQP_API OSQPInt osqp_update_rho(OSQPSolver* solver,
 
 /* ------------------ Derivative stuff ----------------- */
 #ifdef OSQP_ENABLE_DERIVATIVES
-OSQP_API OSQPInt osqp_adjoint_derivative(OSQPSolver*    solver,
-                                         OSQPFloat*     dx,
-                                         OSQPFloat*     dy_l,
-                                         OSQPFloat*     dy_u,
-                                         OSQPCscMatrix* dP,
-                                         OSQPFloat*     dq,
-                                         OSQPCscMatrix* dA,
-                                         OSQPFloat*     dl,
-                                         OSQPFloat*     du);
+OSQP_API OSQPInt osqp_adjoint_derivative_compute(OSQPSolver*    solver,
+                                                 OSQPFloat*     dx,
+                                                 OSQPFloat*     dy_l,
+                                                 OSQPFloat*     dy_u);
+
+OSQP_API OSQPInt osqp_adjoint_derivative_get_mat(OSQPSolver*    solver,
```

Comment:
Do we need these functions to return ints? The code looks to always be returning 0.

Reply:
I think in this case, since the implementation for these derivative functions will likely get more robust with time, it makes sense so that in the future we can return a suitable OSQP_* error under certain conditions, without changing the signature. What do you think?

Reply:
Yea, you are right, this change actually doesn't make sense considering I suggested to add the error check in the comment below this :laughing:. We can leave these returning ints and use it to do error handling.